week3.js

```diff
@@ -0,0 +1,223 @@
+console.log(favoriteAnimals);
+favoriteAnimals.splice(1, 0, "meerkat");
+console.log("using splice() a method that changes the contents of an array by removing or replacing existing elements and/or adding new elements)\n \
+in .splice(1, 0, 'meerkat') the 1 defines the index that i want to work with and 0 defines the number of elements that i want to replace.\n \
+so the value of the array will be:\n \
+[\"blowfish\", \"meerkat\", \"capricorn\", \"giraffe\", \"turtle\"]");
+console.log(favoriteAnimals);
+console.log("The array has a length of: " + favoriteAnimals.length);
+favoriteAnimals.splice(3,1);   // delete giraffe from the previous array
+console.log(favoriteAnimals);
+console.log("The item you are looking for is at index: " + favoriteAnimals.findIndex(animal => animal === "meerkat"));
+
+
+
+
+// More JavaScript
+
+//1
+function sum(x, y, z) {
+  const result = x + y + z;
+  return result;
+}
+
+
+
+//2
+function colorCar(color) {
+	const result = 'a ' + color + ' car';
+	return result;
+}
+
+
+
+//3
+function info(person = {firstName:"mark", lastName:"trump", age:50, eyeColor:"blue"}) {
+	return person;
+}
+
+
+
+//4
+function vehicleType(color, code){
+  if(code === 1){
+  return console.log("a " + color + " car")}
+  else if(code === 2){
+  	return console.log("a " + color + " motorbike")}
+}
+
+
+
+//5
+3 === 3 ? console.log("yes") : console.log("no");
+
+
+
+//6
+function vehicle(color, code, age){
+  if(code === 1){
+  	   if(age > 0){
+  	   	console.log("a " + color + " used car");
+  	   }
+  	   	else{console.log("a " + color + " car");}
+  	   }
+  else if(code === 2){
+  	   if(age > 0){
+  		console.log("a " + color + " used motorbike");
+  	   }
+  		else{
+  	   	console.log("a " + color + " motorbike");}
+  	   }
+}
+
+
+
+//7
+// method 1: using objects (adding another layer of objects for fun!)
+vehicle = {
+	"motor": {
+		id: "motorbike",
+    	id2: "bus",
+     	id3: "truck",
+     	id4: "car"
+     },
+     "railed": {
+     	id: "train",
+     	id2: "tram",
+     },
+     "aircraft": {
+     	id: "airplane",
+     	id2: "helicopter",
+     }
+}
+const firstKey = Object.keys(vehicle)[0];
+const secondKey = Object.keys(vehicle)[1];
+const thirdKey = Object.keys(vehicle)[2];
+//----------------------------------------------------
+console.log(vehicle["aircraft"]["id2"]);
+//logs helicopter
+console.log(vehicle[thirdKey]["id2"]);
+//logs helicopter
+//----------------------------------------------------
+
+//method 2: using arrays
+const vehicles = ["motorbike", "caravan", "bike", "bus"];
+
+
+
+
+//8
+console.log(vehicles[2]);
+
+
+
+
+//9
+const vehicles = ["motorbike", "caravan", "bike", "bus"];
+
+
+function vehicle(color,code,age){
+  let i;
+  if(code === 1){
+  	i = vehicles[0];
+  }
+  else if(code === 2){
+  	i = vehicles[1];
+  }
+  else if(code === 3){
+  	i = vehicles[2]
+  } 
+  else if(code === 4){
+  	i = vehicles[3]
+  } 
```

Comment:
This is fine but you can already see that it becomes a bit cumbersome to write almost the same code over and over again. With larger if statements you can look if a loop is better suited for the job.